extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -0,0 +1,371 @@
+    // TODO(choumx): Only allow binding to attributes allowed by validator.
+
+    const name = attribute.name;
+    if (name.length > 2) {
+      if (name[0] === '[' && name[name.length - 1] === ']') {
+        return {
+          property: name.substr(1, name.length - 2),
+          expression: attribute.value,
+          element,
+        };
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Schedules a vsync task to reevaluate all binding expressions.
+   * @param {boolean=} opt_verifyOnly
+   * @private
+   */
+  digest_(opt_verifyOnly) {
+    // TODO(choumx): Chunk if takes too long for a single frame.
+
+    /** {!VsyncTaskSpecDef} */
+    const task = {measure: this.boundMeasure_, mutate: this.boundMutate_};
+    this.vsync_.run(task, {
+      results: [],
+      verifyOnly: opt_verifyOnly,
+    });
+  }
+
+  /**
+   * Reevaluates all bindings and stores results in `state`.
+   * @param {BindVsyncStateDef} state
+   * @private
+   */
+  measure_(state) {
+    for (let i = 0; i < this.bindings_.length; i++) {
+      const binding = this.bindings_[i];
+      state.results[i] = evaluateBindExpr(binding.expression, this.scope_);
+    }
+  }
+
+  /**
+   * Either applies or verifies the binding evaluation results in `state`.
+   * @param {BindVsyncStateDef} state
+   * @private
+   */
+  mutate_(state) {
+    for (let i = 0; i < this.bindings_.length; i++) {
+      const binding = this.bindings_[i];
+      const result = state.results[i];
+
+      // Don't apply mutation if the result hasn't changed.
+      if (result === this.previousResults_[i]) {
+        continue;
+      } else {
+        this.previousResults_[i] = result;
+      }
+
+      if (state.verifyOnly) {
+        this.verifyBinding_(binding, result);
+      } else {
+        this.applyBinding_(binding, result);
+      }
+    }
+  }
+
+  /**
+   * Applies `newValue` to the element bound in `binding`.
+   * @param {!BindingDef} binding
+   * @param {BindExpressionResultDef} newValue
+   * @private
+   */
+  applyBinding_(binding, newValue) {
+    const property = binding.property;
+    const element = binding.element;
+
+    // TODO(choumx): Does `element.tagName` support binding to `property`?
+
+    // TODO(choumx): Support objects for attributes.
+
+    if (property === 'text') {
+      // TODO(choumx): How to trigger reflow when necessary?
+      element.textContent = newValue;
+    } else if (property === 'class') {
+      // TODO(choumx): SVG elements are an issue, should disallow in validator.
+
+      if (Array.isArray(newValue)) {
+        element.className = newValue.join(' ');
+      } else if (typeof newValue === 'string') {
+        element.className = newValue;
+      } else {
+        user().error(TAG, 'Invalid result for class binding', newValue);
+      }
+    } else {
+      if (newValue === true) {
+        element.setAttribute(property, '');
+      } else if (newValue === false) {
+        element.removeAttribute(property);
```

Comment:
I think the current method can handle that with `null` representing nonexistence and `''` representing an existing boolean attribute.